api/v1beta1/managedclustermodule_types.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ManagedClusterModuleSpec defines the desired state of ManagedClusterModule
+type ManagedClusterModuleSpec struct {
+	// ModuleSpec describes how the KMM operator should deploy a Module on those nodes that need it.
+	ModuleSpec ModuleSpec `json:"moduleSpec,omitempty"`
+
+	// Selector describes on which managed clusters the ModuleSpec should be applied.
+	Selector map[string]string `json:"selector"`
+}
+
+// ManagedClusterModuleStatus defines the observed state of ManagedClusterModule
+type ManagedClusterModuleStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
```

Comment:
What does those 2 means?

Reply:
These 2 `kubebuilder` [markers](https://book.kubebuilder.io/reference/markers.html) were scaffolded by `kubebuilder` when I created the CRD and the reconciler.

The [first](https://book.kubebuilder.io/reference/markers/object.html) one enables interface implementation generation for this type. The [second](https://book.kubebuilder.io/reference/markers/crd.html) enables the `/status` subresource for this type.

Does this make sense?

Reply:
Yup. Thank you for the explanation.